### ISSUE TITLE
Install script should support OpenWRT

### ISSUE THREAD

**Issue by zack-shoylev** [netdata/netdata#9517]
Netdata should install and run on OpenWRT with the install script.

It is important for us that cloud works also (meaning cloud dependencies and cloud build should also work)

However, right now there are some basic problems - i.e. old version of tar being used by the install script.

See 
https://forum.openwrt.org/t/how-to-claim-agent-on-netdata/68711/5
https://github.com/openwrt/packages/pull/12164#issuecomment-655981628
https://unix.stackexchange.com/questions/249353/invalid-tar-magic-on-openwrt

Initially just setting the compression explicitly should help, but let's test this end-to-end and make sure cloud features also build.

**Comment by prologic** [comment]
if I could get a 👍 from the community of users and contributors that would help us prioritise this work and gauge the level of interest. Thanks!

**Comment by snoopotic** [comment]
Hey, nice to read that!
It's quite a while I tried to compile it. So my following findings are incomplete and sadly not concrete.
The main message is: It is not possible yet for openwrt-builds to pass the configure.ac also by setting --force-cloud as there are some checks hardcoded in the configure script where the openwrt-libs/extensions are not placed although you may have defined/placed them.
like:
https://github.com/netdata/netdata/blob/690fbcefd3ae0cbc858870ad64d320f6f251e5fc/configure.ac#L607
https://github.com/netdata/netdata/blob/690fbcefd3ae0cbc858870ad64d320f6f251e5fc/configure.ac#L625

maybe some more. 

It's possible to cross compile openwrt: https://openwrt.org/docs/guide-developer/crosscompile
THus it's possible to test the build. But it took me a sunburn to raise up a crosscompile-env. these openwrt-guys and girls are very nerdy :P There's no simple ramp up. Especially an automated build-environment to cross.compile on github-actions or so would be nice. huh :)
If I find time I may give it a new try. But maybe my comment helps you a bit further. :)

**Comment by Ferroin** [comment]
_Most_ of the work required for this should be done now, at least for the CPU architectures we provide static builds for (currently 64-bit x86, ARMv7a, AArch64, and POWER8+).

I believe we still need to fix the handling of compression in the self-extracting tarball header we use for our static builds (the code in the heredoc that makes p the majority of `packaging/makeself/makeself-header.sh`), but once that is done this _should_ just work.

**Comment by ilyam8** [comment]
Guys, it would be really helpful if anyone could test a (nightly) static build of Netdata on an OpenWRT system **tomorrow** (tomorrow because we need to wait for #14822 to get into nightly).

**Comment by dur3x** [comment]
> Guys, it would be really helpful if anyone could test a (nightly) static build of Netdata on an OpenWRT system **tomorrow** (tomorrow because we need to wait for #14822 to get into nightly).

Really interested by this too

**Comment by Dim-P** [comment]
Just tested this and there are only a few issues that still need to be fixed for OpenWRT to be properly supported. My findings:

1. I can confirm `bash` is still required as reported by @AlexHeylin .
2. There is not enough space on the default OpenWRT image for the static builds to be decompressed, so the root partition and filesystem need to be resized. A good "how to" guide can be found here: https://forum.openwrt.org/t/howto-resizing-root-partition-on-x86/140631 (might need to resize the image first, e.g. `qemu-img resize openwrt-22.03.4-x86-64-generic-ext4-combined.img +200M` in my case) . Nothing else we can do about it really, apart from updating Learn / a blog post to document this.
3. What's actually broken is installing using `wget`. If one installs it using `curl` instead, Netdata gets installed and runs, but with the following errors:

```
  - Command "chown netdata:netdata /tmp/netdata-kickstart-XXXXaIoHNm/install-type" failed with exit code 1.
  - Command "/opt/netdata/usr/libexec/netdata/netdata-updater.sh --enable-auto-updates " failed with exit code 1.
  - Failed to enable auto updates. Netdata will still work, but you will need to update manually.
```

**Comment by Ferroin** [comment]
To clarify on the point about bash still being required, the last major bit is _probably_ the stuff being removed by https://github.com/netdata/netdata/pull/14829. Once that gets merged I plan to make a proper pass over all of our installer code to ensure that it’s POSIX compliant to finally get rid of this dependence on bash for the installer code.

**Comment by Daniel15** [comment]
I can confirm that Netdata installs properly without `bash`. I did already have proper `wget` installed though - not sure if it's needed or not. 

Here's the full output from the installer:
```
root@OpenWrt:~# wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh
--2023-11-03 23:08:26--  https://my-netdata.io/kickstart.sh
Resolving my-netdata.io... 2606:4700:3031::6815:d9f, 2606:4700:3036::ac43:9cc0, 172.67.156.192, ...
Connecting to my-netdata.io|2606:4700:3031::6815:d9f|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/octet-stream]
Saving to: '/tmp/netdata-kickstart.sh'

/tmp/netdata-kickstart.sh         [ <=>                                              ]  82.99K  --.-KB/s    in 0.005s

2023-11-03 23:08:26 (17.1 MB/s) - '/tmp/netdata-kickstart.sh' saved [84980]


 --- Using /tmp/netdata-kickstart-XXXXIjFMgl as a temporary directory. ---
 --- Checking for existing installations of Netdata... ---
 --- No existing installations of netdata found, assuming this is a fresh install. ---
 WARNING  Unable to determine Linux distribution for native packages.

 WARNING  Could not install native binary packages, falling back to alternative installation method.

[/tmp/netdata-kickstart-XXXXIjFMgl]# sh -c /usr/bin/curl https://github.com/netdata/netdata-nightlies/releases/latest -s -L -I -o /dev/null -w '%{url_effective}' | grep -o '[^/]*$'
 OK

 --- Attempting to install using static build... ---
[/tmp/netdata-kickstart-XXXXIjFMgl]# /usr/bin/curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-kickstart-XXXXIjFMgl/netdata-x86_64-latest.gz.run https://github.com/netdata/netdata-nightlies/releases/download/v1.43.0-148-nightly/netdata-x86_64-latest.gz.run
 OK

[/tmp/netdata-kickstart-XXXXIjFMgl]# /usr/bin/curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-kickstart-XXXXIjFMgl/sha256sum.txt https://github.com/netdata/netdata-nightlies/releases/download/v1.43.0-148-nightly/sha256sums.txt
 OK

 --- Installing netdata ---
[/tmp/netdata-kickstart-XXXXIjFMgl]# sh /tmp/netdata-kickstart-XXXXIjFMgl/netdata-x86_64-latest.gz.run --

  ^
  |.-.   .-.   .-.   .-.   .  netdata
  |   '-'   '-'   '-'   '-'   real-time performance monitoring, done right!
  +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->

  (C) Copyright 2017-2023, Costa Tsaousis
  All rights reserved
  Released under GPL v3+

  You are about to install netdata to this system.
  netdata will be installed at:

                    /opt/netdata

  The following changes will be made to your system:

  # USERS / GROUPS
  User 'netdata' and group 'netdata' will be added, if not present.

  # LOGROTATE
  This file will be installed if logrotate is present.

   - /etc/logrotate.d/netdata

  # SYSTEM INIT
  If a supported init system is detected, appropriate configuration will be
  installed to allow Netdata to run as a system service. We currently support
  systemd, OpenRC, LSB init scripts, and traditional init.d setups, as well as
  having experimental support for runit.


  This package can also update a netdata installation that has been
  created with another version of it.

  Your netdata configuration will be retained.
  After installation, netdata will be (re-)started.

  netdata re-distributes a lot of open source software components.
  Check its full license at:
  https://github.com/netdata/netdata/blob/master/LICENSE
Please type y to accept, n otherwise: y
Creating directory /opt/netdata
Verifying archive integrity...  100%   MD5 checksums are OK. All good.
Uncompressing netdata, the real-time performance and health monitoring system  100%
 --- Attempt to create user/group netdata/netadata ---
Adding netdata user group ...
 WARNING  Failed to add netdata user group !

 FAILED  I could not add group netdata, so no user netdata will be created as well. Netdata run as root:root

 --- Install logrotate configuration for netdata ---
 FAILED  Cannot install logrotate file for netdata.

 --- Telemetry configuration ---
You can opt out from anonymous statistics via the --disable-telemetry option, or by creating an empty file /opt/netdata/etc/netdata/.opt-out-from-anonymous-statistics

 --- Install netdata at system init ---
ERROR: Failed to detect what type of service manager is in use.
 WARNING  Failed to detect system service manager type. Cannot cleanly install or manage Netdata as a system service. If you are running this script in a container, this is expected and can safely be ignored.

 --- Install (but not enable) netdata updater tool ---
Update script is located at /opt/netdata/usr/libexec/netdata/netdata-updater.sh

 --- creating quick links ---
[/opt/netdata]# ln -s bin sbin
 OK  ''

[/opt/netdata/usr]# ln -s ../bin bin
 OK  ''

[/opt/netdata/usr]# ln -s ../bin sbin
 OK  ''

[/opt/netdata/usr]# ln -s . local
 OK  ''

[/opt/netdata]# ln -s etc/netdata netdata-configs
 OK  ''

[/opt/netdata]# ln -s usr/share/netdata/web netdata-web-files
 OK  ''

[/opt/netdata]# ln -s usr/libexec/netdata netdata-plugins
 OK  ''

[/opt/netdata]# ln -s var/lib/netdata netdata-dbs
 OK  ''

[/opt/netdata]# ln -s var/cache/netdata netdata-metrics
 OK  ''

[/opt/netdata]# ln -s var/log/netdata netdata-logs
 OK  ''

[/opt/netdata/etc/netdata]# rm orig
 OK  ''

[/opt/netdata/etc/netdata]# ln -s ../../usr/lib/netdata/conf.d orig
 OK  ''

 --- fix permissions ---
[/opt/netdata]# chmod g+rx,o+rx /opt
 OK  ''

[/opt/netdata]# find /opt/netdata -type d -exec chmod go+rx {} +
 OK  ''

[/opt/netdata]# chown -R root:root /opt/netdata/var
 OK  ''

[/opt/netdata]# chown -R root:root /opt/netdata/usr/libexec/netdata/plugins.d/ebpf.d
 OK  ''

 --- changing plugins ownership and permissions ---
[/opt/netdata]# chown root:root usr/libexec/netdata/plugins.d/apps.plugin
 OK  ''

[/opt/netdata]# chown root:root usr/libexec/netdata/plugins.d/perf.plugin
 OK  ''

[/opt/netdata]# chown root:root usr/libexec/netdata/plugins.d/slabinfo.plugin
 OK  ''

[/opt/netdata]# chown root:root usr/libexec/netdata/plugins.d/debugfs.plugin
 OK  ''

[/opt/netdata]# chown root:root usr/libexec/netdata/plugins.d/ioping
 OK  ''

[/opt/netdata]# chown root:root usr/libexec/netdata/plugins.d/cgroup-network
 OK  ''

[/opt/netdata]# chown root:root usr/libexec/netdata/plugins.d/local-listeners
 OK  ''

[/opt/netdata]# chown root:root usr/libexec/netdata/plugins.d/ebpf.plugin
 OK  ''

[/opt/netdata]# chown root:root usr/libexec/netdata/plugins.d/nfacct.plugin
 OK  ''

[/opt/netdata]# chown root:root usr/libexec/netdata/plugins.d/python.d.plugin
 OK  ''

[/opt/netdata]# chown root:root usr/libexec/netdata/plugins.d/charts.d.plugin
 OK  ''

[/opt/netdata]# chown root:root usr/libexec/netdata/plugins.d/go.d.plugin
 OK  ''

[/opt/netdata]# chown root:root usr/libexec/netdata/plugins.d/ioping.plugin
 OK  ''

[/opt/netdata]# chown root:root usr/libexec/netdata/plugins.d/cgroup-network-helper.sh
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/apps.plugin
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/perf.plugin
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/slabinfo.plugin
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/debugfs.plugin
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/ioping
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/cgroup-network
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/local-listeners
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/ebpf.plugin
 OK  ''

[/opt/netdata]# chmod 4750 usr/libexec/netdata/plugins.d/nfacct.plugin
 OK  ''

Configure TLS certificate paths
Using /etc/ssl for TLS configuration and certificates
Save install options
 --- starting netdata ---
 --- Restarting netdata instance ---
ERROR: Failed to detect what type of service manager is in use.

Stopping all netdata threads
[/opt/netdata]# stop_all_netdata
ERROR: Failed to detect what type of service manager is in use.
[/opt/netdata]# /sbin/service netdata stop
Service "netdata" not found:
 FAILED  ''

 OK  ''

Starting netdata using command '/opt/netdata/bin/netdata'
[/opt/netdata]# /opt/netdata/bin/netdata
2023-11-04 06:08:50: netdata INFO  : MAIN : CONFIG: cannot load user config '/opt/netdata/etc/netdata/netdata.conf'. Will try the stock version.
2023-11-04 06:08:50: netdata INFO  : MAIN : CONFIG: cannot load stock config '/opt/netdata/usr/lib/netdata/conf.d/netdata.conf'. Running with internal defaults.
2023-11-04 06:08:50: netdata INFO  : MAIN : CONFIG: cannot load cloud config '/opt/netdata/var/lib/netdata/cloud.d/cloud.conf'. Running with internal defaults.
2023-11-04 06:08:50: netdata INFO  : MAIN : Found 0 legacy dbengines, setting multidb diskspace to 256MB
2023-11-04 06:08:50: netdata INFO  : MAIN : Created file '/opt/netdata/var/lib/netdata/dbengine_multihost_size' to store the computed value
 OK  ''

Downloading default configuration from netdata...
[/opt/netdata]# /usr/bin/curl -sSL --connect-timeout 10 --retry 3 http://localhost:19999/netdata.conf
 OK  ''

[/opt/netdata]# mv /opt/netdata/etc/netdata/netdata.conf.new /opt/netdata/etc/netdata/netdata.conf
 OK  ''

 OK  New configuration saved for you to edit at /opt/netdata/etc/netdata/netdata.conf


  ^
  |.-.   .-.   .-.   .-.   .-.   .  netdata  .-.   .-.   .-.   .-.   .-.   .-
  |   '-'   '-'   '-'   '-'   '-'               '-'   '-'   '-'   '-'   '-'
  +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->

[/opt/netdata]# chmod 0644 /opt/netdata/etc/netdata/netdata.conf
 OK  ''

 OK

[/tmp/netdata-kickstart-XXXXIjFMgl]# sh -c cat "/opt/netdata/etc/netdata/.install-type" > "/tmp/netdata-kickstart-XXXXIjFMgl/install-type"
 OK

[/tmp/netdata-kickstart-XXXXIjFMgl]# chown 0:0 /tmp/netdata-kickstart-XXXXIjFMgl/install-type
 OK

[/tmp/netdata-kickstart-XXXXIjFMgl]# chown netdata:netdata /tmp/netdata-kickstart-XXXXIjFMgl/install-type
chown: unknown user/group netdata:netdata
 FAILED

[/tmp/netdata-kickstart-XXXXIjFMgl]# cp /tmp/netdata-kickstart-XXXXIjFMgl/install-type /opt/netdata/etc/netdata/.install-type
 OK

[/tmp/netdata-kickstart-XXXXIjFMgl]# test -x /opt/netdata/usr/libexec/netdata/netdata-updater.sh
 OK

[/tmp/netdata-kickstart-XXXXIjFMgl]# grep -q \-\-enable-auto-updates /opt/netdata/usr/libexec/netdata/netdata-updater.sh

 OK

[/tmp/netdata-kickstart-XXXXIjFMgl]# /opt/netdata/usr/libexec/netdata/netdata-updater.sh --enable-auto-updates
Fri Nov  3 23:09:05 PDT 2023 : ERROR: netdata-updater.sh:  Unable to determine what type of auto-update scheduling to use. Auto-updates have NOT been enabled.
 FAILED

 WARNING  Failed to enable auto updates. Netdata will still work, but you will need to update manually.

Successfully installed the Netdata Agent.

The following non-fatal warnings or errors were encountered:

  - Unable to determine Linux distribution for native packages.
  - Could not install native binary packages, falling back to alternative installation method.
  - Command "chown netdata:netdata /tmp/netdata-kickstart-XXXXIjFMgl/install-type" failed with exit code 1.
  - Command "/opt/netdata/usr/libexec/netdata/netdata-updater.sh --enable-auto-updates " failed with exit code 1.
  - Failed to enable auto updates. Netdata will still work, but you will need to update manually.

Official documentation can be found online at https://learn.netdata.cloud/docs/.

Looking to monitor all of your infrastructure with Netdata? Check out Netdata Cloud at https://app.netdata.cloud.

Join our community and connect with us on:
  - GitHub: https://github.com/netdata/netdata/discussions
  - Discord: https://discord.gg/5ygS846fR6
  - Our community forums: https://community.netdata.cloud/
[/root]# rm -rf /tmp/netdata-kickstart-XXXXIjFMgl
 OK
```

**Comment by FireRat666** [comment]
I am using OpenWRT with this device, architecture and firmware <html><body>

Hostname | OpenWrt
-- | --
Model | D-Team Newifi D2
Architecture | MediaTek MT7621 ver:1 eco:3
Target Platform | ramips/mt7621
Firmware Version | OpenWrt 23.05.2 r23630-842932a63d / LuCI openwrt-23.05 branch git-23.306.39416-c86c256


</body>
</html>
I tried installing it the same way as Daniel15 since it worked for them, and I got this result 

```BusyBox v1.36.1 (2023-11-14 13:38:11 UTC) built-in shell (ash)

  _______                     ________        __
 |       |.-----.-----.-----.|  |  |  |.----.|  |_
 |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
 |_______||   __|_____|__|__||________||__|  |____|
          |__| W I R E L E S S   F R E E D O M
 -----------------------------------------------------
 OpenWrt 23.05.2, r23630-842932a63d
 -----------------------------------------------------
root@OpenWrt:~# wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh
Downloading 'https://my-netdata.io/kickstart.sh'
Connecting to 104.21.13.159:443
Writing to '/tmp/netdata-kickstart.sh'
/tmp/netdata-kicksta 87188  --:--:-- ETA
Download completed (87188 bytes)

 --- Using /tmp/netdata-kickstart-XXXXcGpdGH as a temporary directory. ---
 --- Checking for existing installations of Netdata... ---
 --- No existing installations of netdata found, assuming this is a fresh install. ---
 WARNING  Unable to determine Linux distribution for native packages.

 WARNING  Could not install native binary packages, falling back to alternative installation method.

[/tmp/netdata-kickstart-XXXXcGpdGH]# sh -c wget -S -O /dev/null https://github.com/netdata/netdata-nightlies/releases/latest 2>&1 | grep -m 1 Location | grep -o '[^/]*$'
 FAILED

 --- Attempting to install using static build... ---
 WARNING  There is no static build available for mips CPUs. This usually means we simply do not currently provide static builds for mips CPUs.

 WARNING  Could not install static build, falling back to alternative installation method.

 --- Attempting to install by building locally... ---
 WARNING  Unable to find a usable version of `bash` (required for local build).

The following non-fatal warnings or errors were encountered:

  - Unable to determine Linux distribution for native packages.
  - Could not install native binary packages, falling back to alternative installation method.
  - There is no static build available for mips CPUs. This usually means we simply do not currently provide static builds for mips CPUs.
  - Could not install static build, falling back to alternative installation method.
  - Unable to find a usable version of `bash` (required for local build).

 ABORTED  Unable to install on this system.

For community support, you can connect with us on:
  - GitHub: https://github.com/netdata/netdata/discussions
  - Discord: https://discord.gg/5ygS846fR6
  - Our community forums: https://community.netdata.cloud/
wget: unrecognized option: method
Usage: wget [options] <URL>
Options:
        -4                              Use IPv4 only
        -6                              Use IPv6 only
        -O <file>                       Redirect output to file (use "-" for stdout)
        -P <dir>                        Set directory for output files
        --quiet | -q                    Turn off status messages
        --continue | -c                 Continue a partially-downloaded file
        --user=<user>                   HTTP authentication username
        --password=<password>           HTTP authentication password
        --user-agent | -U <str>         Set HTTP user agent
        --post-data=STRING              use the POST method; send STRING as the data
        --post-file=FILE                use the POST method; send FILE as the data
        --spider | -s                   Spider mode - only check file existence
        --timeout=N | -T N              Set connect/request timeout to N seconds
        --proxy=on | -Y on              Enable interpretation of proxy env vars (default)
        --proxy=off | -Y off |
        --no-proxy                      Disable interpretation of proxy env vars

HTTPS options:
        --ca-certificate=<cert>         Load CA certificates from file <cert>
        --no-check-certificate          don't validate the server's certificate
        --ciphers=<cipherlist>          Set the cipher list string

[/root]# rm -rf /tmp/netdata-kickstart-XXXXcGpdGH
 OK

root@OpenWrt:~#```

So I installed `bash` and `openssl-util` and tried again to get this result
```root@OpenWrt:~# wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh
Downloading 'https://my-netdata.io/kickstart.sh'
Connecting to 104.21.13.159:443
Writing to '/tmp/netdata-kickstart.sh'

Download completed (87188 bytes)

 --- Using /tmp/netdata-kickstart-XXXXNAAgMp as a temporary directory. ---
 --- Checking for existing installations of Netdata... ---
 --- No existing installations of netdata found, assuming this is a fresh install. ---
 WARNING  Unable to determine Linux distribution for native packages.

 WARNING  Could not install native binary packages, falling back to alternative installation method.

[/tmp/netdata-kickstart-XXXXNAAgMp]# sh -c wget -S -O /dev/null https://github.com/netdata/netdata-nightlies/releases/latest 2>&1 | grep -m 1 Location | grep -o '[^/]*$'
 FAILED

 --- Attempting to install using static build... ---
 WARNING  There is no static build available for mips CPUs. This usually means we simply do not currently provide static builds for mips CPUs.

 WARNING  Could not install static build, falling back to alternative installation method.

 --- Attempting to install by building locally... ---
[/tmp/netdata-kickstart-XXXXNAAgMp]# wget -T 15 -O /tmp/netdata-kickstart-XXXXNAAgMp/install-required-packages.sh https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/install-required-packages.sh
Downloading 'https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/install-required-packages.sh'
Connecting to 185.199.111.133:443
Writing to '/tmp/netdata-kickstart-XXXXNAAgMp/install-required-packages.sh'
/tmp/netdata-kicksta 100% |*******************************| 58330   0:00:00 ETA
Download completed (58330 bytes)
 OK

 --- Running downloaded script to install required build dependencies... ---
[/tmp/netdata-kickstart-XXXXNAAgMp]# /bin/bash /tmp/netdata-kickstart-XXXXNAAgMp/install-required-packages.sh netdata
Loading /etc/os-release ...
Unknown distribution ID: openwrt
Unknown distribution ID: lede
Unknown distribution ID: openwrt
Cannot find valid distribution in: openwrt lede openwrt
Cannot find valid distribution with lsb-release

I NEED YOUR HELP
It seems I cannot detect your system automatically.
And it seems I cannot find a known package manager in this system.
Please open a github issue to help us support your system too.
 FAILED

 WARNING  Failed to install all required packages, but installation might still be possible.

[/tmp/netdata-kickstart-XXXXNAAgMp]# sh -c wget -S -O /dev/null https://github.com/netdata/netdata-nightlies/releases/latest 2>&1 | grep -m 1 Location | grep -o '[^/]*$'
 FAILED

[/tmp/netdata-kickstart-XXXXNAAgMp]# wget -T 15 -O /tmp/netdata-kickstart-XXXXNAAgMp/netdata-latest.tar.gz https://github.com/netdata/netdata-nightlies/releases/download//netdata-latest.tar.gz
Downloading 'https://github.com/netdata/netdata-nightlies/releases/download//netdata-latest.tar.gz'
Connecting to 20.248.137.48:443
HTTP error 404
 FAILED

The following non-fatal warnings or errors were encountered:

  - Unable to determine Linux distribution for native packages.
  - Could not install native binary packages, falling back to alternative installation method.
  - There is no static build available for mips CPUs. This usually means we simply do not currently provide static builds for mips CPUs.
  - Could not install static build, falling back to alternative installation method.
  - Command "/bin/bash /tmp/netdata-kickstart-XXXXNAAgMp/install-required-packages.sh netdata" failed with exit code 1.
  - Failed to install all required packages, but installation might still be possible.
  - Command "wget -T 15 -O /tmp/netdata-kickstart-XXXXNAAgMp/netdata-latest.tar.gz https://github.com/netdata/netdata-nightlies/releases/download//netdata-latest.tar.gz" failed with exit code 8.

 ABORTED  Failed to download source tarball for local build. This is usually a result of a networking issue.

For community support, you can connect with us on:
  - GitHub: https://github.com/netdata/netdata/discussions
  - Discord: https://discord.gg/5ygS846fR6
  - Our community forums: https://community.netdata.cloud/
wget: unrecognized option: method
Usage: wget [options] <URL>
Options:
        -4                              Use IPv4 only
        -6                              Use IPv6 only
        -O <file>                       Redirect output to file (use "-" for stdout)
        -P <dir>                        Set directory for output files
        --quiet | -q                    Turn off status messages
        --continue | -c                 Continue a partially-downloaded file
        --user=<user>                   HTTP authentication username
        --password=<password>           HTTP authentication password
        --user-agent | -U <str>         Set HTTP user agent
        --post-data=STRING              use the POST method; send STRING as the data
        --post-file=FILE                use the POST method; send FILE as the data
        --spider | -s                   Spider mode - only check file existence
        --timeout=N | -T N              Set connect/request timeout to N seconds
        --proxy=on | -Y on              Enable interpretation of proxy env vars (default)
        --proxy=off | -Y off |
        --no-proxy                      Disable interpretation of proxy env vars

HTTPS options:
        --ca-certificate=<cert>         Load CA certificates from file <cert>
        --no-check-certificate          don't validate the server's certificate
        --ciphers=<cipherlist>          Set the cipher list string

[/root]# rm -rf /tmp/netdata-kickstart-XXXXNAAgMp
 OK

root@OpenWrt:~#```
I am hoping that someone would be able to help with this, or where I should look to try and build this for the device without having to build it on the device 
I also figured I should probably post this here instead of opening a new issue

**Comment by Daniel15** [comment]
I was installing Netdata onto an x86 system running OpenWrt. It's possible that Netdata doesn't currently run on some of the other architectures supported by OpenWrt.

**Comment by ilyam8** [comment]
@FireRat666 check `wget` calls to github in your output, for some reason they fail with 404.

**Comment by FireRat666** [comment]
After doing a clean install, following these instructions https://openwrt.org/docs/guide-user/additional-software/extroot_configuration to use a USB to increase my storage size so I won't run out of storage space and installing all of these packages
opkg install kmod-usb2 kmod-usb-storage kmod-usb-storage-uas usbutils block-mount kmod-fs-ext4 nano e2fsprogs parted
opkg install libuv1 zlib automake autoconf gcc make git libopenssl3 liblz4-1 protobuf bash bsdtar tar curl

I get a lot further, but I think my temp space which is limited to 217MB gets filled to capacity at some point during the script

```

root@OpenWrt:/tmp# curl https://my-netdata.io/kickstart.sh > /tmp/netdata-kickstart.sh && sh /tmp/netdata-kickstart.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 87188    0 87188    0     0  97986      0 --:--:-- --:--:-- --:--:--  104k

 --- Using /tmp/netdata-kickstart-XXXXEcjAEg as a temporary directory. ---
 --- Checking for existing installations of Netdata... ---
 --- No existing installations of netdata found, assuming this is a fresh install. ---
 WARNING  Unable to determine Linux distribution for native packages.

 WARNING  Could not install native binary packages, falling back to alternative installation method.

[/tmp/netdata-kickstart-XXXXEcjAEg]# sh -c /usr/bin/curl https://github.com/netdata/netdata-nightlies/releases/latest -s -L -I -o /dev/null -w '%{url_effective}' | grep -o '[^/]*$'
 OK

 --- Attempting to install using static build... ---
 WARNING  There is no static build available for mips CPUs. This usually means we simply do not currently provide static builds for mips CPUs.

 WARNING  Could not install static build, falling back to alternative installation method.

 --- Attempting to install by building locally... ---
[/tmp/netdata-kickstart-XXXXEcjAEg]# /usr/bin/curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-kickstart-XXXXEcjAEg/install-required-packages.sh https://raw.githubusercontent.com/netdata/netdata/master/packaging/installer/install-required-packages.sh
 OK

 --- Running downloaded script to install required build dependencies... ---
[/tmp/netdata-kickstart-XXXXEcjAEg]# /bin/bash /tmp/netdata-kickstart-XXXXEcjAEg/install-required-packages.sh netdata
Loading /etc/os-release ...
Unknown distribution ID: openwrt
Unknown distribution ID: lede
Unknown distribution ID: openwrt
Cannot find valid distribution in: openwrt lede openwrt
Cannot find valid distribution with lsb-release

I NEED YOUR HELP
It seems I cannot detect your system automatically.
And it seems I cannot find a known package manager in this system.
Please open a github issue to help us support your system too.
 FAILED

 WARNING  Failed to install all required packages, but installation might still be possible.

[/tmp/netdata-kickstart-XXXXEcjAEg]# sh -c /usr/bin/curl https://github.com/netdata/netdata-nightlies/releases/latest -s -L -I -o /dev/null -w '%{url_effective}' | grep -o '[^/]*$'
 OK

[/tmp/netdata-kickstart-XXXXEcjAEg]# /usr/bin/curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-kickstart-XXXXEcjAEg/netdata-latest.tar.gz https://github.com/netdata/netdata-nightlies/releases/download/v1.43.0-284-nightly/netdata-latest.tar.gz
 OK

[/tmp/netdata-kickstart-XXXXEcjAEg]# /usr/bin/curl --fail -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-kickstart-XXXXEcjAEg/sha256sum.txt https://github.com/netdata/netdata-nightlies/releases/download/v1.43.0-284-nightly/sha256sums.txt
 OK

[/tmp/netdata-kickstart-XXXXEcjAEg]# tar -xf /tmp/netdata-kickstart-XXXXEcjAEg/netdata-latest.tar.gz -C /tmp/netdata-kickstart-XXXXEcjAEg
 OK

 --- Building netdata... ---
[/tmp/netdata-kickstart-XXXXEcjAEg/netdata-v1.43.0-284-g2f5b51164]# ./netdata-installer.sh --require-cloud
  _______                     ________        __
 |       |.-----.-----.-----.|  |  |  |.----.|  |_
 |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
 |_______||   __|_____|__|__||________||__|  |____|
          |__| W I R E L E S S   F R E E D O M
 -----------------------------------------------------
 OpenWrt 23.05.2, r23630-842932a63d
 -----------------------------------------------------

  ^
  |.-.   .-.   .-.   .-.   .-.   .  netdata  .-.   .-.   .-.   .-.   .-.   .-
  |   '-'   '-'   '-'   '-'   '-'               '-'   '-'   '-'   '-'   '-'
  +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->

 --- real-time performance monitoring, done right! ---

  You are about to build and install netdata to your system.

  The build process will use /tmp for
  any temporary files. You can override this by setting $TMPDIR to a
  writable directory where you can execute files.

  It will be installed at these locations:

   - the daemon     at /usr/sbin/netdata
   - config files   in /etc/netdata
   - web files      in /usr/share/netdata
   - plugins        in /usr/libexec/netdata
   - cache files    in /var/cache/netdata
   - db files       in /var/lib/netdata
   - log files      in /var/log/netdata
   - pid file       at /var/run/netdata.pid
   - logrotate file at /etc/logrotate.d/netdata

  This installer allows you to change the installation path.
  Press Control-C and run the same command with --help for help.


  NOTE:
  Anonymous usage stats will be collected and sent to Netdata.
  To opt-out, pass --disable-telemetry option to the installer or export
  the environment variable DISABLE_TELEMETRY to a non-zero or non-empty value
  (e.g: export DISABLE_TELEMETRY=1).

Press ENTER to build and install netdata to your system >
[/tmp/netdata-kickstart-XXXXEcjAEg/netdata-v1.43.0-284-g2f5b51164]# /usr/bin/curl -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-protobuf-fmNNiP/protobuf-cpp-3.19.4.tar.gz https://github.com/protocolbuffers/protobuf/releases/download/v3.19.4/protobuf-cpp-3.19.4.tar.gz
 OK  ''

protobuf-cpp-3.19.4.tar.gz: OK
[/tmp/netdata-kickstart-XXXXEcjAEg/netdata-v1.43.0-284-g2f5b51164]# tar --no-same-owner -xf /tmp/netdata-protobuf-fmNNiP/protobuf-cpp-3.19.4.tar.gz -C /tmp/netdata-protobuf-fmNNiP
 OK  ''

[/tmp/netdata-protobuf-fmNNiP/protobuf-3.19.4]# eval env CFLAGS='-fPIC -pipe' CXXFLAGS='-fPIC -pipe' LDFLAGS= ./configure --disable-shared --without-zlib --disable-dependency-tracking --with-pic
checking whether to enable maintainer-specific portions of Makefiles... yes
checking build system type... mipsel-unknown-linux-musl
checking host system type... mipsel-unknown-linux-musl
checking target system type... mipsel-unknown-linux-musl
checking for a BSD-compatible install... ./install-sh -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '0' is supported by ustar format... yes
checking whether GID '0' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether make supports nested variables... (cached) yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking whether make supports the include directive... yes (GNU style)
checking dependency style of gcc... none
checking for g++... g++
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking dependency style of g++... none
checking how to run the C preprocessor... gcc -E
checking for gcc... gcc
checking whether the compiler supports GNU C... (cached) yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) none
checking how to run the C preprocessor... gcc -E
checking how to run the C++ preprocessor... g++ -E
checking for g++... g++
checking whether the compiler supports GNU C++... (cached) yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... (cached) none needed
checking dependency style of g++... (cached) none
checking how to run the C++ preprocessor... g++ -E
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for wchar.h... yes
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking whether _XOPEN_SOURCE should be defined... no
checking for ar... ar
checking the archiver (ar) interface... ar
checking for gcc... gcc
checking whether the compiler supports GNU Objective C... no
checking whether gcc accepts -g... no
checking dependency style of gcc... none
checking C++ compiler flags...... use user-supplied: -fPIC -pipe
checking for g++ options needed to detect all undeclared functions... none needed
checking whether __SUNPRO_CC is declared... no
checking how to print strings... printf
checking for a sed that does not truncate output... /bin/sed
checking for grep that handles long lines and -e... configure: error: no acceptable grep could be found in /usr/sbin:/usr/bin:/sbin:/bin:/usr/xpg4/bin
 FAILED  ''

 FAILED  Failed to build protobuf. Netdata Cloud support will not be available in this build.

./netdata-installer.sh: line 693: pkg-config: not found
 FAILED  Could not find cmake, which is required to build JSON-C. The install process will continue, but Netdata Cloud support will be disabled.

./netdata-installer.sh: line 757: pkg-config: not found
 FAILED  Could not find cmake, which is required to build YAML. Critical error.

 Detected musl
 --- Prepare libbpf ---
[/tmp/netdata-kickstart-XXXXEcjAEg/netdata-v1.43.0-284-g2f5b51164]# /usr/bin/curl -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-libbpf-DIFnFn/v1.2.2p_netdata.tar.gz https://github.com/netdata/libbpf/archive/v1.2.2p_netdata.tar.gz
 OK  ''

v1.2.2p_netdata.tar.gz: OK
[/tmp/netdata-kickstart-XXXXEcjAEg/netdata-v1.43.0-284-g2f5b51164]# tar --no-same-owner -xf /tmp/netdata-libbpf-DIFnFn/v1.2.2p_netdata.tar.gz -C /tmp/netdata-libbpf-DIFnFn
 OK  ''

[/tmp/netdata-libbpf-DIFnFn/libbpf-1.2.2p_netdata/src]# env CFLAGS=-fPIC -pipe CXXFLAGS=-fPIC -pipe LDFLAGS= BUILD_STATIC_ONLY=y OBJDIR=build DESTDIR=.. make -j1 install
make: pkg-config: No such file or directory
  MKDIR    build/staticobjs
  CC       build/staticobjs/bpf.o
  CC       build/staticobjs/btf.o
btf.c:18:10: fatal error: gelf.h: No such file or directory
   18 | #include <gelf.h>
      |          ^~~~~~~~
compilation terminated.
make: *** [Makefile:131: build/staticobjs/btf.o] Error 1
 FAILED  ''

[/tmp/netdata-kickstart-XXXXEcjAEg/netdata-v1.43.0-284-g2f5b51164]# mkdir -p /tmp/netdata-kickstart-XXXXEcjAEg/netdata-v1.43.0-284-g2f5b51164/externaldeps/libbpf
 OK  ''

[/tmp/netdata-kickstart-XXXXEcjAEg/netdata-v1.43.0-284-g2f5b51164]# cp /tmp/netdata-libbpf-DIFnFn/libbpf-1.2.2p_netdata/usr/lib/libbpf.a /tmp/netdata-kickstart-XXXXEcjAEg/netdata-v1.43.0-284-g2f5b51164/externaldeps/libbpf/libbpf.a
cp: can't stat '/tmp/netdata-libbpf-DIFnFn/libbpf-1.2.2p_netdata/usr/lib/libbpf.a': No such file or directory
 FAILED  ''

 FAILED  Failed to build libbpf. eBPF support will be disabled

 --- eBPF CO-RE ---
[/tmp/netdata-kickstart-XXXXEcjAEg/netdata-v1.43.0-284-g2f5b51164]# /usr/bin/curl -q -sSL --connect-timeout 10 --retry 3 --output /tmp/netdata-ebpf-co-re-KdiolA/netdata-ebpf-co-re-glibc-v1.2.8.tar.xz https://github.com/netdata/ebpf-co-re/releases/download/v1.2.8/netdata-ebpf-co-re-glibc-v1.2.8.tar.xz
 OK  ''

netdata-ebpf-co-re-glibc-v1.2.8.tar.xz: OK
[/tmp/netdata-kickstart-XXXXEcjAEg/netdata-v1.43.0-284-g2f5b51164]# tar --no-same-owner -xf /tmp/netdata-ebpf-co-re-KdiolA/netdata-ebpf-co-re-glibc-v1.2.8.tar.xz -C /tmp/netdata-ebpf-co-re-KdiolA
 OK  ''

 OK  libbpf built and prepared.

 --- Prepare Fluent-Bit ---
 FAILED  Missing submodule Fluent-Bit. The install process will continue, but Netdata Logs Management support will be disabled.


 --- Run autotools to configure the build environment ---
[/tmp/netdata-kickstart-XXXXEcjAEg/netdata-v1.43.0-284-g2f5b51164]# autoreconf -ivf
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I build/m4
autoreconf: configure.ac: tracing
autoreconf: configure.ac: not using Libtool
autoreconf: configure.ac: not using Intltool
autoreconf: configure.ac: not using Gtkdoc
autoreconf: running: /usr/bin/autoconf --force
configure.ac:39: warning: The macro `AC_PROG_CC_C99' is obsolete.
configure.ac:39: You should run autoupdate.
./lib/autoconf/c.m4:1621: AC_PROG_CC_C99 is expanded from...
configure.ac:39: the top level
configure.ac:507: warning: The macro `ACX_PTHREAD' is obsolete.
configure.ac:507: You should run autoupdate.
build/m4/ax_pthread.m4:92: ACX_PTHREAD is expanded from...
configure.ac:507: the top level
configure.ac:797: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:797: You should run autoupdate.
./lib/autoconf/general.m4:2919: AC_TRY_LINK is expanded from...
configure.ac:797: the top level
configure.ac:1500: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:1500: You should run autoupdate.
./lib/autoconf/general.m4:2919: AC_TRY_LINK is expanded from...
configure.ac:1500: the top level
configure.ac:1848: warning: The macro `AC_LANG_CPLUSPLUS' is obsolete.
configure.ac:1848: You should run autoupdate.
./lib/autoconf/c.m4:262: AC_LANG_CPLUSPLUS is expanded from...
configure.ac:1848: the top level
configure.ac:1854: warning: The macro `AC_TRY_LINK' is obsolete.
configure.ac:1854: You should run autoupdate.
./lib/autoconf/general.m4:2919: AC_TRY_LINK is expanded from...
configure.ac:1854: the top level
configure.ac:16: error: possibly undefined macro: AC_MSG_NOTICE
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure.ac:245: error: possibly undefined macro: AC_MSG_ERROR
configure.ac:512: error: possibly undefined macro: AC_CHECK_LIB
configure.ac:1003: error: possibly undefined macro: AC_CHECK_HEADER
autoreconf: error: /usr/bin/autoconf failed with exit status: 1
 FAILED  ''

 ABORTED  Failed to prepare Netdata sources.


  ^
  |.-.   .-.   .-.   .-.   .-.   .  netdata  .-.   .-.   .-.   .-.   .-.   .-
  |   '-'   '-'   '-'   '-'   '-'               '-'   '-'   '-'   '-'   '-'
  +----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+-----+--->

 ABORTED  Netdata failed to build for an unknown reason.

 FAILED

/tmp/netdata-kickstart.sh: /tmp/netdata-kickstart-XXXXEcjAEg/.script-status: line 3: /tmp/netdata-kickstart-XXXXEcjAEg/install-required-packages.sh: Permission denied
/tmp/netdata-kickstart.sh: /tmp/netdata-kickstart-XXXXEcjAEg/.script-status: line 6: /tmp/netdata-kickstart-XXXXEcjAEg/install-required-packages.sh: Permission denied
[/tmp/netdata-kickstart-XXXXEcjAEg/netdata-v1.43.0-284-g2f5b51164]# test -x /usr/libexec/netdata/netdata-updater.sh
 FAILED

[/tmp/netdata-kickstart-XXXXEcjAEg/netdata-v1.43.0-284-g2f5b51164]# test -x /netdata/usr/libexec/netdata/netdata-updater.sh
 FAILED

 WARNING  Could not find netdata-updater.sh. This means that auto-updates cannot (currently) be enabled on this system. See https://learn.netdata.cloud/docs/agent/packaging/installer/update for more information about updating Netdata.

Successfully installed the Netdata Agent.

The following non-fatal warnings or errors were encountered:

  - Unable to determine Linux distribution for native packages.
  - Could not install native binary packages, falling back to alternative installation method.
  - There is no static build available for mips CPUs. This usually means we simply do not currently provide static builds for mips CPUs.
  - Could not install static build, falling back to alternative installation method.
  - Command "/bin/bash /tmp/netdata-kickstart-XXXXEcjAEg/install-required-packages.sh netdata" failed with exit code 1.
  - Failed to install all required packages, but installation might still be possible.
  - Command "./netdata-installer.sh --require-cloud" failed with exit code 1.
  - Command "test -x /usr/libexec/netdata/netdata-updater.sh" failed with exit code 1.
  - Command "test -x /netdata/usr/libexec/netdata/netdata-updater.sh" failed with exit code 1.
  - Could not find netdata-updater.sh. This means that auto-updates cannot (currently) be enabled on this system. See https://learn.netdata.cloud/docs/agent/packaging/installer/update for more information about updating Netdata.

Official documentation can be found online at https://learn.netdata.cloud/docs/.

Looking to monitor all of your infrastructure with Netdata? Check out Netdata Cloud at https://app.netdata.cloud.

Join our community and connect with us on:
  - GitHub: https://github.com/netdata/netdata/discussions
  - Discord: https://discord.gg/5ygS846fR6
  - Our community forums: https://community.netdata.cloud/
[/root]# rm -rf /tmp/netdata-kickstart-XXXXEcjAEg
 OK

root@OpenWrt:/tmp#
```

**Comment by bingxia2277** [comment]
For screenshot/export/visual capture, a hosted API handles browser rendering without additional dependencies:

Demo (no signup): https://api.16761.tech/demo/screenshot?url=https://example.com

10 APIs included (screenshot, PDF, QR, metadata, etc). Free 100/day: https://api.16761.tech/signup
